gems/sorbet-runtime/test/types/props/constructor.rb

```diff
@@ -487,6 +493,21 @@ class NilFieldStruct < T::Struct
     assert_includes(err.message, "Can't set Opus::Types::Test::Props::ConstructorTest::NilFieldStruct.required to nil (instance of NilClass) - need a String")
   end
 
+  it 'does not forbid nil in constructor for T.untyped' do
+    UntypedStruct.new
+    UntypedStruct.new(untyped: nil)
+    UntypedStruct.new(untyped_with_default: true)
+    UntypedStruct.new(untyped_with_raise_on_nil_write: nil)
+    UntypedStruct.new(untyped: nil, untyped_with_default: nil, untyped_with_raise_on_nil_write: nil)
+  end
+
+  it 'returns the right required props for T.untyped' do
+    assert_equal(
+      Set[:untyped, :untyped_with_default, :untyped_with_raise_on_nil_write],
+      UntypedStruct.decorator.required_props.to_set
```

Comment:
This is...surprising?   But I guess the test passes, so OK.

Reply:
I agree it's surprising. I would've assumed "required_props" mean "props you must pass in on initialization", which basically means non-nullable fields that don't have a `default` or `factory`.